### ISSUE TITLE
Doesnt seem to work on 2017.3.0f3

### ISSUE THREAD

**Issue by slymatt** [Bunny83/UnityWindowsFileDrag-Drop#1]
Hello, 

Could do with some help it doesn't seem to work on unity version 2017.3.0f3 as you drag something over the top you get the stop sign.

Also checked in 2018.1.0f2, is there anything more that needs to be done?

Thanks in advance for help

**Comment by Bunny83** [comment]
I just tried it in Unity 2018.1.0b11 and it just works the same as before. Have you added the "FileDragAndDrop" component to a gameobject? Though it's only meant as example usage.

I've tested on 64bit Window10 and it works in the editor as well as in a standalone build. You haven't added any details to reproduce the issue.

**Comment by slymatt** [comment]
Hello

the issue is only in the editor and you reproduce it by just adding the script to a gamrobjet.

Not sure why it works for you but not me I tried it on a fresh project too and that's acted the same as I drag over the unity editor my cursor changes to a gray stop sign

Best regards

On Sat, 12 May 2018, 08:40 Bunny83, <notifications@github.com<mailto:notifications@github.com>> wrote:

I just tried it in Unity 2018.1.0b11 and it just works the same as before. Have you added the "FileDragAndDrop" component to a gameobject? Though it's only meant as example usage.

I've tested on 64bit Window10 and it works in the editor as well as in a standalone build. You haven't added any details to reproduce the issue.

**Comment by Bunny83** [comment]
Note that it was never meant to be used inside the editor. Inside the editor you could have undocked the gameview or even multiple gameviews. This framework only pick one native window and accept drag and drop events. Also keep in mind that you have to be in playmode in the editor.

Note that there is a strange bug which could be the result of an incompatibility with Unity's native drag and drop handler in the editor. When you unfocus and refocus the editor and then drag a file onto the editor, the editor can randomly crash silently / close itself without any notice.

Though, even it's not meant for in editor use, [it does work for me](https://www.dropbox.com/s/7iv4ttglc41juhf/DragAndDropExample.gif?raw=1)

**Comment by slymatt** [comment]
Yes still doesn't seem to work as you can see when you drag over the text file you get the plus icon I don't get that. I'll try screen cap for you soon so you can see why happens for me
On Sat, 12 May 2018, 15:48 Bunny83, <notifications@github.com<mailto:notifications@github.com>> wrote:

Note that it was never meant to be used inside the editor. Inside the editor you could have undocked the gameview or even multiple gameviews. This framework only pick one native window and accept drag and drop events. Also keep in mind that you have to be in playmode in the editor.

Note that there is a strange bug which could be the result of an incompatibility with Unity's native drag and drop handler in the editor. When you unfocus and refocus the editor and then drag a file onto the editor, the editor can randomly crash silently / close itself without any notice.

Though, even it's not meant for in editor use, it does work for me<https://www.dropbox.com/s/7iv4ttglc41juhf/DragAndDropExample.gif?raw=1>

**Comment by keyserjaya** [comment]
Hello Bunny83 i have the same issue too, maybe an example scene/project from you could help :)

**Comment by JulienAtWork** [comment]
Just wanted to comment that i had the same problem; but then i discovered that it was actually a unity/windows problem. If unity is started as admin it doesn't accept drag and drop from non-admin apps as the explorer. Maybe this info can help someone. 
Also, crashing has stopped for me after i replaced WinAPI.CallNextHookEx(...) with IntPtr.Zero in the Callback (i'm not sure, but i guess it just prevents other (non existent) hooks from running)

**Comment by Bian-Sh** [comment]
I have the same issue too ,I am using unity 2017.3.1f1 ,and can not drag and drop into Editor even it is in play mode.
and  as you've say it never meant to use in editor , I just mean to mention it .

and another thing which confuse me that much is ,  when I Login windows using a non-Admin-group User ,then I run the app that I have already build up ,but drag &drop can not work , even I right-click on it and select the “Run as Administrator” ,but still not work .

and I debug the event ,it showed that the hook emit so many event ,such as : "MOUSEWHEEL"、"NCMOUSEMOVE" and so on, but not show "DROPFILES" .

and the good news is that it works well when i login windows use Admin-Group user . 

sorry for my weak language skill.

see ↓，thk~
https://github.com/Bian-Sh/DragFile4unityRT

**Comment by Bunny83** [comment]
This is not an issue with unity but a general security "feature" of Windows. No drag and drop operations are allowed between application that have different privileges. Make sure you don't run unity as administrator if you use a normal user.

**Comment by JulienAtWork** [comment]
Just an update: my previous attempt to fix the crashes by leaving out CallNextHookEx seemed to have been only temporary, but currently i haven't had a crash yet since i separated my handling logic from the drop hook callback (i just put the list of files into a queue then dequeue it in LateUpdate and do the work there), even with CallNextHookEx still in there.

**Comment by ritor1sefa** [comment]
https://youtu.be/Bhaw-L_xxlw
At first, something didn’t work either.
In the editor - does not work.
"DragFile4unityRT-master" tried - it runs endlessly and closes itself

Test it on Unity 2017.2.0f3 (64-bit) and in Unity 2018.2.11f1 (64-bit)
On windows 10(17763.1) 64
https://www.sendspace.com/file/sy5ii4 (2017.2.0f3 unity version project folder)
